### DISCUSSION TITLE
which port should FlightSQL handler use

### DISCUSSION THREAD

**Discussion by youngsofun** [databendlabs/databend#10741] [General]
# FlightSQL

50050:

https://arrow.apache.org/ballista/user-guide/flightsql.html
[Using FlightSQL to Connect to Ballista — Arrow DataFusion documentation (apache.org)](https://arrow.apache.org/ballista/user-guide/flightsql.html)

example server in arrow_rs  use 50051

31337:

https://voltrondata.com/resources/running-arrow-flight-server-querying-data-jdbc-adbc
[Running an Arrow Flight SQL Server and Querying Data with JDBC and ADBC | Voltron Data](https://voltrondata.com/resources/running-arrow-flight-server-querying-data-jdbc-adbc)

https://github.com/voltrondata/flight-duckdb-example
[voltrondata/flight-duckdb-example: Sample code to accompany blog post showcasing Arrow Flight SQL running on DuckDB (github.com)](https://github.com/voltrondata/flight-duckdb-example)

## other JDBC port
![image](https://user-images.githubusercontent.com/5782159/227196078-03a01f07-8690-40de-95e4-027ee4ab8c28.png)

https://www.doc.ic.ac.uk/~pjm/databases/jdbc.html
[JDBC (ic.ac.uk)](https://www.doc.ic.ac.uk/~pjm/databases/jdbc.html)

**Comment by youngsofun** [comment]
50050 report `Address already in use` sometimes

https://github.com/datafuselabs/databend/actions/runs/4499896016/jobs/7918538998

**Reply by Xuanwo** [reply]
https://en.wikipedia.org/wiki/Ephemeral_port

> Many [Linux kernels](https://en.wikipedia.org/wiki/Linux_kernel) use the port range 32768–60999.

I'm guessing is't better to use lower port.

**Reply by Xuanwo** [reply]
How about using `7000` as default flight sql handler port?

**Comment by youngsofun** [comment]
```
➜  ~ lsof -i tcp:7000
COMMAND   PID        USER   FD   TYPE             DEVICE SIZE/OFF NODE NAME
ControlCe 569 yangxiufeng    5u  IPv4 0x161698f94c923ffd      0t0  TCP *:afs3-fileserver (LISTEN)
ControlCe 569 yangxiufeng    6u  IPv6 0x161698f949caa465      0t0  TCP *:afs3-fileserver (LISTEN)
```

https://techshift.net/what-is-afs3-fileserver/

**Comment by youngsofun** [comment]
according to https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers
8600-8628 is not used
so I let`s try 8600 

1. start cluster in ci: at least n  n+2 n +3 not used
2. 8600 and 8000 both http and starts with 8

----
8600 is used in stateful standalone
